### PR TITLE
Adds the API for Metrics logging

### PR DESCRIPTION
**Pull request by dushyanthsc**
Change only adds the Metrics class that extends Artifact and defines the API the class provides.

The API will be implemented once Artifact class supports the Metadata based Artifact schemas.

### CODE REVIEW

File: sdk/python/kfp/dsl/metrics.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Metric classes in MLMD ontology for KFP SDK."""
+
+from typing import Any, Dict, List
+
+from kfp.dsl import artifact
+
+class Metrics(artifact.Artifact):
+  """Supports capturing summary metrics for a model from components such as ModelEvaluation."""
```
Comment (chensun) [inline sdk/python/kfp/dsl/metrics.py:21]: nit: this is very optional, if not too much trouble, can you please make sure the line doesn't exceed 80 characters?

File: sdk/python/kfp/dsl/metrics.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  def log_precision(self, value: float) -> None:
+    """Logs Model's precision metric."""
+    pass
+
+  def log_metric(self, metric: str, value: float) -> None:
+    """Log a single custom metric key-value pair"""
+    pass
+
+
+class ClassificationMetrics(artifact.Artifact):
```
Comment (numerology) [inline sdk/python/kfp/dsl/metrics.py:42]: Just for my education, why `ClassificationMetrics` inherits `Artifact` but not `Metrics`?

File: sdk/python/kfp/dsl/schemas/classification_metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
+title: ClassificationEvaluationMetrics
```
Comment (chensun) [inline sdk/python/kfp/dsl/schemas/classification_metrics.yaml:1]: I don't see these yaml files being referenced in code, but when you do need these, they should be listed in https://github.com/kubeflow/pipelines/blob/master/sdk/python/MANIFEST.in

**Comment by dushyanthsc** [comment]
retiring the PR in favor of https://github.com/kubeflow/pipelines/pull/5259

**Pull request closed without merging**